Makefile

```diff
@@ -266,6 +268,11 @@ minikube-deploy: minikube-start gadget-default-container kubectl-gadget
 btfgen:
 	+make -f Makefile.btfgen
 
+.PHONY: check_build_requirements
+check_build_requirements:
+	@which go > /dev/null && go version | grep $(GOLANG_VERSION) || echo 'You need to have golang version $(GOLANG_VERSION) to build kubectl_gadget' >&2
+	@which docker > /dev/null && docker buildx --help > /dev/null || echo 'You need to have docker and docker buildx installed to build ig and container images' >&2
+
```

Comment:
I'm not sure if this is worth it to check those things automatically. Perhaps we can only document them and that's it? (It could be more complicated to things versions of installed libraries and so on in case they're needed).

Reply:
The check could be done outside of a rule so that it is always executed. This is already done for the `pv` command. We could do something like:

```
GOLANG_VERSION=1.19

ifneq (0, $(shell go version | grep -q $(GOLANG_VERSION) &>/dev/null; echo $$?))
$(error "go $(GOLANG_VERSION) not available")
endif

ifneq (0, $(shell docker buildx --help &>/dev/null; echo $$?))
$(error "docker buildx not available")
endif
```

Reply:
> This is already done for the pv

IIUC the logic we have is to use the pv command if this is available. It's not a strict requirement.

I'm still not convinced about this kind of checks. Is there really an advantage of printing such warnings over letting the command fail:

```
$ make gadget-default-container
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
if false == "true" ; then \
	./tools/getbtfhub.sh && \
	make -f Makefile.btfgen BPFTOOL=/root/btfhub/tools/bin/bpftool. \
		BTFHUB_ARCHIVE=/root/btfhub-archive/ OUTPUT=hack/btfs/ -j; \
fi
docker buildx build --load -t ghcr.io/inspektor-gadget/inspektor-gadget:latest \
	-f Dockerfiles/gadget-default.Dockerfile .
make: docker: No such file or directory
make: *** [Makefile:142: gadget-default-container] Error 127
```

Also, I think those warnings are misleading as they'll be printed even if the target doesn't require them:

```
$ make btfgen
/bin/sh: 1: go: not found
Makefile:49: "You need to have golang version go1.19 installed to build kubectl_gadget"
/bin/sh: 1: docker: not found
Makefile:53: 'You need to have docker and docker buildx installed to build ig and container images'
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
make: go: No such file or directory
make -f Makefile.btfgen
...
```